### PR TITLE
fix: rfox stake confirm handle not enough fee asset for gas

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Handles the case of bridging FOX into Arb while not having enough ETH.ARB for gas.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes N/A, spotted by ops in https://discord.com/channels/554694662431178782/1252410161163931730/1255024199706083338

Low

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Have enough ETH.ARB for gas at RFOX stake confirm step and confirm there's no regression here
- Send your whole balance out of your account 0 (e.g with MM to ensure no dust remains) and confirm you now see "Not enough ETH.ARB to cover gas" with a disabled button

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

Before/after sending the whole account 0 ETH.ARB balance out of the account:

https://jam.dev/c/cf67d65e-a661-4fb0-89a7-ba5d04122c44

### CODE REVIEW

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (woodenfurniture) [inline src/pages/RFOX/components/Stake/StakeConfirm.tsx:70]: suggestion:

```suggestion
    selectFeeAssetByChainId(state, stakingAsset.chainId),
```

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (woodenfurniture) [inline src/pages/RFOX/components/Stake/StakeConfirm.tsx:189]: question: should we be summing the approval fee with the staking fee so a user is never able to get into a situation where they only have enough balance to approve but after that not enough to stake?

Comment (woodenfurniture) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:189]: e.g 
```typescript
const totalNetworkFeeCryptoBaseUnit = bnOrZero(approvalFees?.totalNetworkFeeCryptoBaseUnit)
  .plus(bnOrZero(stakeFees?.totalNetworkFeeCryptoBaseUnit))
  ```

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (woodenfurniture) [inline src/pages/RFOX/components/Stake/StakeConfirm.tsx:297]: actually loving the early returns like this, much more pleasing than `} else if () {` everywhere

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (gomesalexandre) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:70]: would be noicer but then we'd have to opt-chain to empty string :(
image

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (gomesalexandre) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:189]: Was originally going to say absolutely! Unfortunately, we can't get Tx fees if an approval is required, because getting Tx fees means simulating the Tx, and the Tx would revert on approval needed.
So `bnOrZero(stakeFees)` would always evaluate to 0 in `bnOrZero(approvalFees).plus(bnOrZero(stakeFees))` if an approval is required

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (woodenfurniture) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:70]: these selectors man..

File: src/pages/RFOX/components/Stake/StakeConfirm.tsx
Comment (woodenfurniture) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:189]: Makes sense, actually we should comment that!

Comment (woodenfurniture) [reply src/pages/RFOX/components/Stake/StakeConfirm.tsx:189]:
```suggestion
    // Unfortunately, we can't get Tx fees if an approval is required, because getting Tx fees means simulating the Tx, and the Tx would revert on approval needed.
    // So bnOrZero(stakeFees?.totalNetworkFeeCryptoBaseUnit) would always evaluate to 0 in the expression above, if an approval is required.
    const fees = approvalFees || stakeFees
```

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**
